LeetcodeProblems/Algorithms/Happy_Number.js

```diff
@@ -0,0 +1,49 @@
+/**
+Happy Number
+https://leetcode.com/problems/happy-number/
+
+Write an algorithm to determine if a number n is happy.
+
+A happy number is a number defined by the following process:
+
+    Starting with any positive integer, replace the number by the sum of the squares of its digits.
+    Repeat the process until the number equals 1 (where it will stay), or it loops endlessly in a cycle which does not include 1.
+    Those numbers for which this process ends in 1 are happy.
+
+Return true if n is a happy number, and false if not.
+
+Example 1:
+Input: n = 19
+Output: true
+Explanation:
+12 + 92 = 82
+82 + 22 = 68
+62 + 82 = 100
+12 + 02 + 02 = 1
+
+Example 2:
+Input: n = 2
+Output: false
+
+
+ */
+
+/**
+ * @param {number} n
+ * @return {boolean}
+ */
+var isHappy = function(n) {
+    return checkHappyNumber(n);
+};
+
+function checkHappyNumber(n){
+    strNumber = n.toString();
+    splitNumber = strNumber.split("");
```

Comment:
another solution that doesn't require translating the integer to a String would be to 
get the number, divide it by 10, get the digit and repeat the process.

Reply:
sorry, i don't get it. could you give an example?

Reply:
```
last_digit = n % 10 
n = (n - last_digit) / 10
```